makedep.py fails on some machines due to a unicode character

Due to [this snippet](https://github.com/cp2k/cp2k/commit/d3d36bb8a17094ec894bde01ec3cab3f23415d1d), I have experienced compilation errors with fresh clones on some machines. the output is

```
make: Entering directory '/scratch/luschr/git/cp2k'
Discovering programs ...
make -C /scratch/luschr/git/cp2k/exts/dbcsr -f .cp2k/Makefile \
   INCLUDEMAKE=/scratch/luschr/git/cp2k/arch/local.psmp \
   LIBDIR=/scratch/luschr/git/cp2k/lib/local/psmp/exts/dbcsr \
   OBJDIR=/scratch/luschr/git/cp2k/obj/local/psmp/exts/dbcsr \
   ACC="" \
   ACCFLAGS=""
Removing stale archives for psmp ...
Removing stale archives ...
Resolving dependencies ...
Resolving dependencies for psmp ...
Traceback (most recent call last):
  File "/scratch/luschr/git/cp2k/tools/build_utils/makedep.py", line 429, in <module>
    src_files=args.src_file,
  File "/scratch/luschr/git/cp2k/tools/build_utils/makedep.py", line 45, in main
    parse_file(parsed_files, fn, src_dir)  # parses also included files
  File "/scratch/luschr/git/cp2k/tools/build_utils/makedep.py", line 203, in parse_file
    content = fhandle.read()
  File "/usr/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 3479: ordinal not in range(128)
/scratch/luschr/git/cp2k/Makefile:448: recipe for target 'makedep' failed
make[2]: *** [makedep] Error 1
/scratch/luschr/git/cp2k/Makefile:126: recipe for target 'psmp' failed
make[1]: *** [psmp] Error 2
Makefile:121: recipe for target 'all' failed
make: *** [all] Error 2
make: Leaving directory '/scratch/luschr/git/cp2k'
```

For now I just changed the character to d, but I am sure we can tell Python to use Unicode decoding?

--- a/tools/build_utils/check_archives.py
+++ b/tools/build_utils/check_archives.py
@@ -3,7 +3,7 @@
 # author: Ole Schuett
 
 import sys, os
-import subprocess
+from subprocess import check_output
 from os import path
 from glob import glob
 
@@ -29,7 +29,7 @@ def main():
     known_archives = set()
     for root, dirs, files in os.walk(src_dir):
         if "PACKAGE" in files:
-            content = open(path.join(root, "PACKAGE")).read()
+            content = open(path.join(root, "PACKAGE"), encoding="utf8").read()
             package = eval(content)
 
             archive = "libcp2k" + path.basename(root)
@@ -46,7 +46,7 @@ def main():
                 parts[0] for parts in file_parts if parts[-1] in KNOWN_EXTENSIONS
             ]
 
-            output = check_output([ar_exe, "t", archive_fn])
+            output = check_output([ar_exe, "t", archive_fn], encoding="utf8")
             for line in output.strip().split("\n"):
                 if line == "__.SYMDEF SORTED":
                     continue  # needed for MacOS
@@ -63,15 +63,6 @@ def main():
     for archive_fn in rogue_archives:
         print("Found rogue archive %s , removing archive." % (archive_fn))
         os.remove(archive_fn)
-
-
-# =============================================================================
-def check_output(*popenargs, **kwargs):
-    """ backport for Python 2.4 """
-    p = subprocess.Popen(stdout=subprocess.PIPE, *popenargs, **kwargs)
-    output = p.communicate()[0]
-    assert p.wait() == 0
-    return output.decode()
 
 
 # =============================================================================

--- a/tools/build_utils/discover_programs.py
+++ b/tools/build_utils/discover_programs.py
@@ -39,7 +39,7 @@ def main():
 
 # ============================================================================
 def is_fortran_program(fn):
-    f = open(fn)
+    f = open(fn, encoding="utf8")
     s = path.getsize(fn)
     f.seek(max(0, s - 100))
     tail = f.read()
@@ -50,7 +50,7 @@ def is_fortran_program(fn):
 
 # ============================================================================
 def has_main_function(fn):
-    f = open(fn)
+    f = open(fn, encoding="utf8")
     content = f.read()
     f.close()
     m = re_main.search(content)

--- a/tools/build_utils/makedep.py
+++ b/tools/build_utils/makedep.py
@@ -181,7 +181,7 @@ $(EXEDIR)/{bfn}.$(ONEVERSION) : {bfn}.o {deps}
 
         makefile += "\n\n"
 
-    with open(out_fn, "w") as fhandle:
+    with open(out_fn, "w", encoding="utf8") as fhandle:
         fhandle.write(makefile)
         fhandle.close()
 
@@ -199,7 +199,7 @@ def parse_file(parsed_files, fn, src_dir):
     if fn in parsed_files:
         return
 
-    with open(fn) as fhandle:
+    with open(fn, encoding="utf8") as fhandle:
         content = fhandle.read()
 
     # re.IGNORECASE is horribly expensive. Converting to lower-case upfront
@@ -261,7 +261,7 @@ def read_pkg_manifest(project_name, packages, p):
     if not path.exists(fn):
         error("Could not open PACKAGE manifest: " + fn)
 
-    with open(fn) as fhandle:
+    with open(fn, encoding="utf8") as fhandle:
         content = fhandle.read()
 
     packages[p] = ast.literal_eval(content)

--- a/tools/conventions/analyze_gfortran_ast.py
+++ b/tools/conventions/analyze_gfortran_ast.py
@@ -220,5 +220,5 @@ in the cp2k arch-file.
     for fn in args.files:
         assert fn.endswith(".ast")
 
-        with open(fn) as fhandle:
+        with open(fn, encoding="utf8") as fhandle:
             process_log_file(fhandle)

--- a/tools/conventions/analyze_gfortran_warnings.py
+++ b/tools/conventions/analyze_gfortran_warnings.py
@@ -122,7 +122,7 @@ def may_call_implicit(loc, routine):
 
     pkg = path.dirname(loc)
     manifest_fn = pkg + "/PACKAGE"
-    with open(manifest_fn) as fhandle:
+    with open(manifest_fn, encoding="utf8") as fhandle:
         manifest = ast.literal_eval(fhandle.read())
 
     if "implicit" not in manifest:
@@ -152,5 +152,5 @@ in the cp2k arch-file.
     args = parser.parse_args()
 
     for fn in args.files:
-        with open(fn) as fhandle:
+        with open(fn, encoding="utf8") as fhandle:
             check_warnings(fhandle)

--- a/tools/conventions/analyze_src.py
+++ b/tools/conventions/analyze_src.py
@@ -76,7 +76,7 @@ def validate(cp2k_dir, filelist=None, excluded_dirs=DEFAULT_EXCLUDED_DIRS):
 
     # also exclude any Git submodules
     if path.exists(path.join(cp2k_dir, ".gitmodules")):
-        with open(path.join(cp2k_dir, ".gitmodules"), "r") as fhandle:
+        with open(path.join(cp2k_dir, ".gitmodules"), encoding="utf8") as fhandle:
             excluded_dirs += [
                 path.join(cp2k_dir, l.split()[-1]) for l in fhandle if "path =" in l
             ]
@@ -99,7 +99,7 @@ def validate(cp2k_dir, filelist=None, excluded_dirs=DEFAULT_EXCLUDED_DIRS):
             if fn_ext in ("template", "instantiate"):
                 continue
 
-            with open(path.join(root, fn)) as fhandle:
+            with open(path.join(root, fn), encoding="utf8") as fhandle:
                 content = fhandle.read()
 
             # check banner
@@ -137,9 +137,9 @@ def validate(cp2k_dir, filelist=None, excluded_dirs=DEFAULT_EXCLUDED_DIRS):
 
     flags = [f for f in flags if not flag_exceptions_re.match(f)]
 
-    with open(path.join(cp2k_dir, "INSTALL.md")) as fhandle:
+    with open(path.join(cp2k_dir, "INSTALL.md"), encoding="utf8") as fhandle:
         install_txt = fhandle.read()
-    with open(path.join(cp2k_dir, "src/cp2k_info.F")) as fhandle:
+    with open(path.join(cp2k_dir, "src/cp2k_info.F"), encoding="utf8") as fhandle:
         cp2k_info = fhandle.read()
 
     flags_src = re.search(

--- a/tools/conventions/summarize_issues.py
+++ b/tools/conventions/summarize_issues.py
@@ -11,7 +11,7 @@ def summarize(issue_files, suppressions):
     suppress = []
 
     if suppressions:
-        with open(suppressions) as fhandle:
+        with open(suppressions, encoding="utf8") as fhandle:
             suppress = (line.rstrip() for line in fhandle)
             # ignore empty and commented out lines
             suppress = [line for line in suppress if line and not line.startswith("#")]
@@ -19,7 +19,7 @@ def summarize(issue_files, suppressions):
     issues = []
 
     for fn in issue_files:
-        with open(fn) as fhandle:
+        with open(fn, encoding="utf8") as fhandle:
             lines = (line.strip() for line in fhandle)
             # only add non-empty lines
             issues += [line for line in lines if line]

